### ISSUE TITLE
How do I configure multiple eclipse che instances on one aws EC2 instance for multiple users (<10) with nginx?

### ISSUE THREAD

**Issue by tequa** [eclipse-che/che#4067]
I've read through many issues related to this, e.g. issue 877 and many of the nginx issues, but I'm still a little confused.

So what I want is an interface where each user has their own secure eclipse che instance with their own workspaces on the same EC2 instance. From what I've seen, nginx is the best way to do this because you can have multiple urls map to the same ip (my EC2 instance) through different ports. And I think I'm correct that through nginx I can choose which ips have access to which ports? So if I have user "tequa" with his own eclipse che instance, the idea is that only his machine will have access to *.amazon.com/tequa (which maps to port xxxx), I'm assuming?

I installed nginx on the ec2 instance and edited the nginx.conf file so that my two different che folders (with the instances and workspaces) were reverse proxied to two different local host ports. But I feel like this is not at all what I should be doing since these folders just have data and not the actual docker images for eclipse che? 

So how do I set this reverse proxy for the eclipse che docker containers? I saw your article where you did this with workspaces, but I'm not sure how to do this with the whole eclipse che container.

**Comment by TylerJewell** [comment]
While some in the community are working on certain types of configurations, this is not something that has been published or certified.  At this time, we recommend that you consider deploying Codenvy "docker run codenvy/cli start" which is a version of Eclipse Che that has authentication, scaling, and built in port routing into it.

We may enable this sort of support in the future for Che, but today it would require a significant configuration of nginx, and it would not be secure. While nginx could block certain access to the che server, it would not be possible to block access to every workspace's terminal + ws agent + user-servers inside of it. The fact that you want to give each user their own URL on the front end, which could map to different workpaces on the back-end just makes the problem more challenging.

Of course, we welcome the community to document any nginx configurations that they provide.  If Codenvy is not a good option for you right now, I would recommend running a separate che server for each of your users. You can bind different Che servers to different ports, and so each user would get their own dedicated IP + port combination.

**Comment by tequa** [comment]
Thanks Tyler - so you've mentioned on a few posts that one eclipse che instance with a workspace for each user would work better. If I did this, for security would it work just to configure the specific >36000 ports for each workspace to be only accessible to only the user in mind? And then make the dashboard (8080 port) accessible for only the team?

**Comment by TylerJewell** [comment]
There are a few things that are going on:
1. Che server (port 8080), but this will vary for each che server instance.

2. Each workspace will be on some random ports starting at 32768.  You will not be able to predict it because each ws container will have its ports assigned by Docker.  So it will be first come, first serve.  Every time a ws starts, a terminal is opened, or one of your users starts a server in their workspace, this binds to an internal container port and then the Docker daemon will expose it to a unique port starting at 32768.  So you will have to handled this somehow.

Even if you have many Che servers running, they will all be sharing a single Docker daemon on the same host, so will have these port conflicts.  It took us a team of 10 people a couple years to figure out a lot of this stuff within Codenvy.

I think what we will eventually do for che is hide everything behind a single port, and then embed a proxy that handles routing of all traffic, so that admins only have to open and configure a single port.  It'll take us a couple quarters to engineer that unfortunately.

**Comment by tequa** [comment]
I'll give Codenvy a try then - this issue can be closed. Thanks a bunch, appreciate your help.

**Comment by tequa** [comment]
So I removed all eclipse/che containers and started a codenvy/cli instance:

docker run -it --rm -v /var/run/docker.sock:/var/run/docker.sock -v /home/ubuntu/codenvy/:/data codenvy/cli start

It started and said: (codenvy start): Use: http://$ip:80. The codenvy/agents:5.2.2 container was mapped to 9000/tcp. I tried accessing the ports of all the containers but either didn't get anything (this included the 9000 port, which is open) or got a blank page.

I removed all of these containers and tried codenvy/che (I'm not sure what's the difference. Should I be using this?), using my data folder from my previous eclipse/che instance. This worked fine although I'm not sure what the difference is between this and eclipse/che.

**Comment by TylerJewell** [comment]
@tequa - your original start syntax looks ok.  What happened when you got to `http://it:80`.  Were you able to login as admin and then create workspaces?

I wouldn't be looking at the other containers and ports that we are opening up - those are (mostly) for our internal management by Codenvy.

Definitely don't use `codenvy/che`.  :)

**Comment by tequa** [comment]
I got nothing. I checked my ports and here's the port 80 entry: 

27301/docker-proxy
tcp6       0      0 :::80                   :::*                    LISTEN

**Comment by TylerJewell** [comment]
You will need to provide a "codenvy/cli info" and "codenvy/cli --bundle".  There are not enough details to diagnose any issues.  Could be firewall, could be networking configuration, could be all sorts of stuff on your system that is interfering.

**Comment by tequa** [comment]
Sorry for delayed response - it was an EC2 instance problem and works fine now. After talking it over with the team, we want development environments that are completely hosted from the EC2 servers. So Codenvy is probably not the right way to go since the workspaces are hosted from your end, correct? I'm thinking what you mentioned earlier, an eclipse che server and IP:port for each user, is the best way to go.

**Comment by TylerJewell** [comment]
You can run Codenvy on EC2.  Codenvy is just a scalable version of Che -
installs exactly the same way (though you cannot change the port).  You can
start with "docker run codenvy/cli start".  There is a three user license
for free built into it for testing purposes.


Tyler Jewell | CEO | tyler@​codenvy.​com | 9​78​.8​84​.53​55

**Comment by tequa** [comment]
Alright, my mistake - so I had started a Codenvy instance and tried to create an account but it said "Your admin has not accepted the license agreement". So I went to this page: https://codenvy.com/product/pricing/ and created an account and once on the dashboard believed I was running off of my EC2 instance, which of course I wasn't. After not seeing any workspace docker containers running on the instance, I misunderstood how Codenvy works.

Anyway - every time I've started a codenvy instance, I've accepted the license agreement so I'm not sure why this is happening. I followed your steps from here but still had the issue: https://github.com/codenvy/codenvy/issues/1576

$docker run -it --rm -v /var/run/docker.sock:/var/run/docker.sock -v /home/ubuntu/codenvy2:/data codenvy/cli:5.1.1 init

...
...
_INFO: (codenvy init): Do you accept the Codenvy license? (https://codenvy.com/legal/fair-source/)

      I accept the license: [Y/n] Y

INFO: (codenvy init): Installing configuration and bootstrap variables:
INFO: (codenvy init):   CODENVY_HOST=$host_ip
INFO: (codenvy init):   CODENVY_VERSION=5.1.1
INFO: (codenvy init):   CODENVY_CONFIG=/home/ubuntu/codenvy2
INFO: (codenvy init):   CODENVY_INSTANCE=/home/ubuntu/codenvy2/instance_

$ docker run -it --rm -v /var/run/docker.sock:/var/run/docker.sock -v /home/ubuntu/codenvy2:/data codenvy/cli:5.1.1 start

_WARNING: No swap limit support
INFO: (codenvy cli): Checking network... (hint: '--fast' skips version, network, and nightly checks)
INFO: (codenvy cli): Loading cli...
INFO: (codenvy cli): Checking registry for version '5.1.1' images
INFO: (codenvy config): Generating codenvy configuration...
INFO: (codenvy config): Customizing docker-compose for running in a container
INFO: (codenvy start): Preflight checks
         port 80 (http):        [AVAILABLE]
         port 443 (https):      [AVAILABLE]
         port 2181 (zookeeper): [AVAILABLE]
         port 5000 (registry):  [AVAILABLE]
         port 23750 (socat):    [AVAILABLE]
         port 23751 (swarm):    [AVAILABLE]
         port 32001 (jmx):      [AVAILABLE]
         port 32101 (jmx):      [AVAILABLE]

INFO: (codenvy start): Starting containers...
INFO: (codenvy start): Services booting...
INFO: (codenvy start): Server logs at "docker logs -f codenvy_codenvy_1"
INFO: (codenvy start): Booted and reachable
INFO: (codenvy start): Ver: 5.1.1
INFO: (codenvy start): Use: $host_ip
INFO: (codenvy start): API: $host_ip/swagger_

Logs:
_Waiting for Postgres to boot...
psql: could not connect to server: Connection refused
        Is the server running on host "postgres" (172.21.0.3) and accepting
        TCP/IP connections on port 5432?
[INFO]: Postgres is unavailable. Trying in 5 seconds...
                                 List of databases
   Name    |  Owner   | Encoding |  Collate   |   Ctype    |   Access privileges
-----------+----------+----------+------------+------------+-----------------------
 dbcodenvy | postgres | UTF8     | en_US.utf8 | en_US.utf8 |
 postgres  | postgres | UTF8     | en_US.utf8 | en_US.utf8 |
 template0 | postgres | UTF8     | en_US.utf8 | en_US.utf8 | =c/postgres          +
           |          |          |            |            | postgres=CTc/postgres
 template1 | postgres | UTF8     | en_US.utf8 | en_US.utf8 | =c/postgres          +
           |          |          |            |            | postgres=CTc/postgres
(4 rows)

[INFO]: Postgres is up and running. Starting Codenvy Tomcat...
2017-02-10 19:09:28,935[main]             [INFO ] [o.a.c.s.VersionLoggerListener 89]    - Server version:        Apache Tomcat/8.0.38
2017-02-10 19:09:28,956[main]             [INFO ] [o.a.c.s.VersionLoggerListener 91]    - Server built:          Oct 6 2016 20:51:55 UTC
2017-02-10 19:09:28,976[main]             [INFO ] [o.a.c.s.VersionLoggerListener 93]    - Server number:         8.0.38.0
2017-02-10 19:09:28,977[main]             [INFO ] [o.a.c.s.VersionLoggerListener 95]    - OS Name:               Linux
2017-02-10 19:09:28,977[main]             [INFO ] [o.a.c.s.VersionLoggerListener 97]    - OS Version:            4.4.0-53-generic
2017-02-10 19:09:28,978[main]             [INFO ] [o.a.c.s.VersionLoggerListener 99]    - Architecture:          amd64
2017-02-10 19:09:28,978[main]             [INFO ] [o.a.c.s.VersionLoggerListener 101]   - Java Home:             /opt/jdk1.8.0_102/jre
2017-02-10 19:09:28,979[main]             [INFO ] [o.a.c.s.VersionLoggerListener 103]   - JVM Version:           1.8.0_102-b14
2017-02-10 19:09:28,980[main]             [INFO ] [o.a.c.s.VersionLoggerListener 105]   - JVM Vendor:            Oracle Corporation
2017-02-10 19:09:28,995[main]             [INFO ] [o.a.c.s.VersionLoggerListener 107]   - CATALINA_BASE:         /opt/codenvy-tomcat
2017-02-10 19:09:28,996[main]             [INFO ] [o.a.c.s.VersionLoggerListener 109]   - CATALINA_HOME:         /opt/codenvy-tomcat
2017-02-10 19:09:28,998[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Djava.util.logging.config.file=/opt/codenvy-tomcat/conf/logging.properties
2017-02-10 19:09:28,998[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager
2017-02-10 19:09:28,999[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Xms512m
2017-02-10 19:09:29,010[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Xmx2048m
2017-02-10 19:09:29,011[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -XX:+HeapDumpOnOutOfMemoryError
2017-02-10 19:09:29,012[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -XX:+UseConcMarkSweepGC
2017-02-10 19:09:29,012[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -XX:+CMSParallelRemarkEnabled
2017-02-10 19:09:29,012[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -XX:+UseCMSInitiatingOccupancyOnly
2017-02-10 19:09:29,013[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -XX:+ScavengeBeforeFullGC
2017-02-10 19:09:29,013[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -XX:+CMSScavengeBeforeRemark
2017-02-10 19:09:29,013[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dche.local.conf.dir=/opt/codenvy-data/conf/
2017-02-10 19:09:29,014[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dauth.handler.default=org
2017-02-10 19:09:29,014[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dche.logs.dir=/opt/codenvy-data/logs
2017-02-10 19:09:29,015[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dche.logs.level=INFO
2017-02-10 19:09:29,015[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Djdk.tls.ephemeralDHKeySize=2048
2017-02-10 19:09:29,015[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Djava.protocol.handler.pkgs=org.apache.catalina.webresources
2017-02-10 19:09:29,016[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dcom.sun.management.jmxremote=true
2017-02-10 19:09:29,016[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Djava.rmi.server.hostname=172.31.9.227
2017-02-10 19:09:29,017[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dcom.sun.management.jmxremote.authenticate=true
2017-02-10 19:09:29,017[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dcom.sun.management.jmxremote.password.file=/opt/codenvy-data/conf/jmxremote.password
2017-02-10 19:09:29,017[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dcom.sun.management.jmxremote.access.file=/opt/codenvy-data/conf/jmxremote.access
2017-02-10 19:09:29,018[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dcom.sun.management.jmxremote.ssl=false
2017-02-10 19:09:29,050[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Djava.endorsed.dirs=/opt/codenvy-tomcat/endorsed
2017-02-10 19:09:29,051[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dcatalina.base=/opt/codenvy-tomcat
2017-02-10 19:09:29,051[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Dcatalina.home=/opt/codenvy-tomcat
2017-02-10 19:09:29,052[main]             [INFO ] [o.a.c.s.VersionLoggerListener 115]   - Command line argument: -Djava.io.tmpdir=/opt/codenvy-tomcat/temp
2017-02-10 19:09:29,679[main]             [INFO ] [o.a.c.http11.Http11NioProtocol 443]  - Initializing ProtocolHandler ["http-nio-8080"]
2017-02-10 19:09:29,732[main]             [INFO ] [o.a.t.util.net.NioSelectorPool 72]   - Using a shared selector for servlet write/read
2017-02-10 19:09:29,736[main]             [INFO ] [o.a.catalina.startup.Catalina 591]   - Initialization processed in 2031 ms
2017-02-10 19:09:29,915[main]             [INFO ] [c.m.JmxRemoteLifecycleListener 332]  - The JMX Remote Listener has configured the registry on port 32001 and the server on port 32101 for the Platform server
2017-02-10 19:09:29,988[main]             [INFO ] [o.a.c.core.StandardService 435]      - Starting service Catalina
2017-02-10 19:09:30,002[main]             [INFO ] [o.a.c.core.StandardEngine 259]       - Starting Servlet Engine: Apache Tomcat/8.0.38
2017-02-10 19:09:30,215[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 911]        - Deploying web application archive /opt/codenvy-tomcat/webapps/site.war
2017-02-10 19:09:53,129[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 975]        - Deployment of web application archive /opt/codenvy-tomcat/webapps/site.war has finished in 22,913 ms
2017-02-10 19:09:53,158[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 911]        - Deploying web application archive /opt/codenvy-tomcat/webapps/api.war
2017-02-10 19:10:33,224[ost-startStop-1]  [INFO ] [o.f.c.i.d.DbSupportFactory 44]       - Database: jdbc:postgresql://codenvy-postgres:5432/dbcodenvy (PostgreSQL 9.5)
2017-02-10 19:10:33,260[ost-startStop-1]  [INFO ] [o.f.c.i.util.VersionPrinter 44]      - Flyway 4.0.3 by Boxfuse
2017-02-10 19:10:33,308[ost-startStop-1]  [INFO ] [o.f.c.i.d.DbSupportFactory 44]       - Database: jdbc:postgresql://codenvy-postgres:5432/dbcodenvy (PostgreSQL 9.5)
2017-02-10 19:10:33,417[ost-startStop-1]  [INFO ] [i.f.CustomSqlMigrationResolver 153]  - Searching for sql scripts in locations [classpath:che-schema, classpath:codenvy-schema]
2017-02-10 19:10:33,484[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbValidate 44]       - Successfully validated 8 migrations (execution time 00:00.077s)
2017-02-10 19:10:33,576[ost-startStop-1]  [INFO ] [o.f.c.i.m.MetaDataTableImpl 44]      - Creating Metadata table: "public"."schema_version"
2017-02-10 19:10:33,676[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Current version of schema "public": << Empty Schema >>
2017-02-10 19:10:33,688[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Migrating schema "public" to version 5.0.0.8.1 - 1__init.sql
2017-02-10 19:10:34,543[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Migrating schema "public" to version 5.0.0.8.1.1 - 1.1__init.sql
2017-02-10 19:10:35,421[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Migrating schema "public" to version 5.0.0.9.1 - 1__add_index_on_workspace_temporary.sql
2017-02-10 19:10:35,523[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Migrating schema "public" to version 5.0.0.9.1.1 - 1.1__rename_managecodenvy_action.sql
2017-02-10 19:10:35,631[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Migrating schema "public" to version 5.0.0.9.1.2 - 1.2__add_organization_distributed_resources.sql
2017-02-10 19:10:35,667[ost-startStop-1]  [WARN ] [o.f.c.i.dbsupport.JdbcTemplate 48]   - DB: identifier "fk_organization_distributed_resources_resource_organization_distributed_resources_id" will be truncated to "fk_organization_distributed_resources_resource_organization_dis" (SQL State: 42622 - Error Code: 0)
2017-02-10 19:10:35,722[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Migrating schema "public" to version 5.0.0.9.2 - 2__update_local_links_in_environments.sql
2017-02-10 19:10:35,813[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Migrating schema "public" to version 5.0.0.9.2.1 - 2.1__add_license_actions.sql
2017-02-10 19:10:35,867[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Migrating schema "public" to version 5.1.0.0.1 - 0.1__remove_sync_agents.sql
2017-02-10 19:10:35,932[ost-startStop-1]  [INFO ] [o.f.c.i.command.DbMigrate 44]        - Successfully applied 8 migrations to schema "public" (execution time 00:02.359s).
2017-02-10 19:10:40,833[ost-startStop-1]  [INFO ] [c.c.a.u.s.AdminUserCreator 98]       - Admin user 'admin' successfully created
2017-02-10 19:10:44,052[ted-scheduler-1]  [INFO ] [ockerAbandonedResourcesCleaner 123]  - List containers registered in the api: []
2017-02-10 19:10:45,228[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 975]        - Deployment of web application archive /opt/codenvy-tomcat/webapps/api.war has finished in 52,070 ms
2017-02-10 19:10:45,229[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 911]        - Deploying web application archive /opt/codenvy-tomcat/webapps/ide-resources.war
2017-02-10 19:10:51,802[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 975]        - Deployment of web application archive /opt/codenvy-tomcat/webapps/ide-resources.war has finished in 6,573 ms
2017-02-10 19:10:51,805[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 911]        - Deploying web application archive /opt/codenvy-tomcat/webapps/mail.war
2017-02-10 19:10:56,382[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 975]        - Deployment of web application archive /opt/codenvy-tomcat/webapps/mail.war has finished in 4,577 ms
2017-02-10 19:10:56,399[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 911]        - Deploying web application archive /opt/codenvy-tomcat/webapps/docs.war
2017-02-10 19:10:57,548[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 975]        - Deployment of web application archive /opt/codenvy-tomcat/webapps/docs.war has finished in 1,148 ms
2017-02-10 19:10:57,549[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 911]        - Deploying web application archive /opt/codenvy-tomcat/webapps/factory.war
2017-02-10 19:11:15,835[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 975]        - Deployment of web application archive /opt/codenvy-tomcat/webapps/factory.war has finished in 18,286 ms
2017-02-10 19:11:15,836[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 911]        - Deploying web application archive /opt/codenvy-tomcat/webapps/swagger.war
2017-02-10 19:11:16,659[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 975]        - Deployment of web application archive /opt/codenvy-tomcat/webapps/swagger.war has finished in 823 ms
2017-02-10 19:11:16,660[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 911]        - Deploying web application archive /opt/codenvy-tomcat/webapps/ROOT.war
2017-02-10 19:11:33,203[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 975]        - Deployment of web application archive /opt/codenvy-tomcat/webapps/ROOT.war has finished in 16,543 ms
2017-02-10 19:11:33,204[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 911]        - Deploying web application archive /opt/codenvy-tomcat/webapps/dashboard.war
2017-02-10 19:11:48,918[ost-startStop-1]  [INFO ] [o.a.c.startup.HostConfig 975]        - Deployment of web application archive /opt/codenvy-tomcat/webapps/dashboard.war has finished in 15,714 ms
2017-02-10 19:11:48,939[main]             [INFO ] [o.a.c.http11.Http11NioProtocol 488]  - Starting ProtocolHandler ["http-nio-8080"]
2017-02-10 19:11:48,984[main]             [INFO ] [o.a.catalina.startup.Catalina 642]   - Server startup in 139247 ms_

**Comment by tequa** [comment]
Oh nevermind, I didn't realize admin and password was literally "admin" and "password" the first time. ¯\_(ツ)_/¯ My mistake.

**Comment by TylerJewell** [comment]
Well, that was entirely entertaining to read.  I've been having a bit of a head cold the past couple days, so I am a little slow on the up take.  But I can see the struggles through the posts that you made.  Glad you got it set up!

**Comment by tequa** [comment]
Thanks, appreciate your help - so after the new Eclipse Che update I just wanted to test out multiple Eclipse Che containers on the same server. The problem is that whenever I start a new container on a different port,  the container on the first port receives a SIGTERM. ("Received SIGTERM. Stopping Che server running on localhost:$port... A valid shutdown command was received via the shutdown port"). This container is then shut down and removed.

Here's the command I'm using:

docker run -it --rm -e CHE_PORT=$PORT -v $pathtodata:/data -v  $pathtoche:/che -v /var/run/docker.sock:/var/run/docker.sock eclipse/che:latest start

I use different $pathtodata and $pathtoche folders for each new Che container I try to run. The new instance always starts successfully and is mapped to the correct port. HOWEVER, what I think is the issue is that the ports of the che container always include both the port I specified as well as the 8000 and 8080 ports. For example, after I started a container on port 8181:

Container: che-8181
Ports: 8000/tcp, 8080/tcp, 0.0.0.0:8181->8181/tcp

and when I start a container on port 8080:
Container: che-8080
Ports: 8000/tcp, 0.0.0.0:8080->8080/tcp

Why are 8000 and 8080 included here? I can include the log but I don't think it contains any pertinent information besides the two line shutdown protocol mentioned above.
